WordPressKit/RemotePost.m

```diff
@@ -47,4 +48,48 @@ - (NSDictionary *)debugProperties {
     return [NSDictionary dictionaryWithDictionary:debugProperties];
 }
 
+/// A hash used to determine if the remote content has changed.
+///
+/// This hash must remain constant regardless of iOS version, app restarts or instances used. `Hasher` or NSObject's `hash` were not used for these reasons.
+///
+/// `dateModified` is not included within the hash as it is prone to change wihout the content having been changed and is the reason this hash is necessary.
+///
+/// `autosave` properties are intentionally omitted as remote autosaves are always discarded in favor of local autosaves (aka `revision`s)
+///
+/// - Note: At the time of writing the backend will occasionally create updates that neither come from autosaves or user initiated saves, these will modify the hash
+/// and at present are treated as genuine updates as they are triggered by the user changing their posts content.
+- (NSString *)contentHash
+{
+    NSString *hashedContents = [NSString stringWithFormat:@"%@/%@/%@%@/%@/%@%@/%@/%@%@/%@/%@%@/%@/%@%@/%@/%@%@/%@/%@%@/%@/%@%@/%@/%@",
```

Comment:
Considering we have support to hash each component in the methods below, and then combine the hash of each component (through `combineHashes`), is there a reason behind the choice of using a big string concatenating all of the post elements?

Reply:
Performance mostly.

Hashing every property will likely minimize collisions but it took ~ 0.1s when I performance tested this. I fear this could result in a significant delay. 

#### Property hash performance
![image](https://user-images.githubusercontent.com/72017236/97746363-7dca2000-1aea-11eb-8d9c-8a06ae07c47d.png)


For an `AbstractPost` upload I don't think this is a problem as a post upload isn't an action that blocks the user. However for downloading a `RemotePost` update this could result in real user delays as the non-arrival of this update will block them from continued use of the app.

#### ConcatenatedStringHash Performance
After concatenating the strings we're down to < 0.01s

![image](https://user-images.githubusercontent.com/72017236/97747383-2c229500-1aec-11eb-8b3d-0a09b38c2ab7.png)

I'm still a little bit concerned that this has the potential to be a bit too slow if we're going to be checking large numbers of posts each update. The swift algorithm I wrote earlier this month just performs a bitwise shift operation, this would be much faster than the others if it prove necessary.

How are posts currently downloaded when in large numbers? Do we download them in batches?